clusterstate/clusterstate.go

```diff
@@ -0,0 +1,130 @@
+package clusterstate
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/e2etests/clusterstate/provider"
+)
+
+type Config struct {
+	Logger   micrologger.Logger
+	Provider provider.Interface
+}
+
+type ClusterState struct {
+	logger   micrologger.Logger
+	provider provider.Interface
+}
+
+func New(config Config) (*ClusterState, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.Provider == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Provider must not be empty", config)
+	}
+
+	s := &ClusterState{
+		logger:   config.Logger,
+		provider: config.Provider,
+	}
+
+	return s, nil
+}
+
+func (c *ClusterState) Test(ctx context.Context) error {
+	var err error
+
+	{
+		c.logger.LogCtx(ctx, "level", "debug", "message", "installing e2e-app")
+
+		err = c.provider.InstallTestApp()
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		c.logger.LogCtx(ctx, "level", "debug", "message", "installed e2e-app")
+	}
+
+	{
+		c.logger.LogCtx(ctx, "level", "debug", "message", "waiting for guest cluster")
+
+		err = c.provider.WaitForGuestReady()
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		c.logger.LogCtx(ctx, "level", "debug", "message", "guest cluster ready")
```

Comment:
Ready test checks the nodes and that the test app is installed.

Reply:
So far the guest ready check is already done in the test setup. Or is this a ready check for the test app?

Reply:
See my other comment. I'll split the check for the e2e app to avoid repeating the nodes ready test.

Reply:
Done.